#1699 - add configurable interval to .hyper.js config file watch.

paulbouwer opened the pull request:
While looking at another issue related to `gaze` scanning the home folder on Windows, I fixed the following: 
- #1699 - Hyper is reading .hyper.js 40 times a second, constantly.

You can update the following in the `.hyper.js` config file to change the interval:

```
    // number of milliseconds to wait between checking if config file has been updated
    configFileWatchInterval: 5000
```

I have verified with ProcessMonitor on Windows, that the watch() read interval has been altered to the configured value.

This PR should be ready to merge.

rauchg commented:
@paulbouwer sounds like we should correct the default. Once every 2 second sounds reasonable?

rauchg commented:
Oops you did that exactly. I think maybe we should lower it to 2?

ppot commented:
This should not be in config. But strictly typed.

rauchg commented:
I think maybe @ppot would be right. If we have a reasonable default, do we need customization?

paulbouwer commented:
@rauchg @ppot Should I simply set the interval to 2s and remove from the config file? Happy to do this - 2s is a better default than what we currently have.

Or is there merit in setting the default to 2s and leaving it in config file for those people who really want to minimise disk access further?

ppot commented:
@paulbouwer Yes you definitely remove it from the config file. For the case of people wanting to change it's value. This can be modified later on if requests move along this way.

paulbouwer commented:
@ppot I've removed it from config and added 2s interval to **gaze** call.

```
function watch() {
  // watch for changes on config every 2s
  gaze(path, {interval: 2000}, function (err) {
```

It should be good to go.

ppot commented:
@paulbouwer As I said: Use a `const scan = 2000;` on the upper

stereokai commented:
I am the one who originally reported #1699 (measured Hyper to be accessing `.hyper.js` *40 times a second*).

It's great to see you guys addressing this issue. I really appreciate that! If I may, I would like to add that looking at the bigger picture, even 2 seconds is highly excessive. Who changes their settings so often?

Personally, I wouldn't even watch the settings file, but require users to manually refresh. Don't gasp yet. A terminal emulator's settings file is not a development environment, where live-reloading justifies the additional I/O burden.

ppot commented:
@stereokai Yes, reducing it to each min would be a better idea. Else user can manually refresh.

stereokai commented:
@paulbouwer @ppot is it possible to make this setting opt-out via config like you suggested at the [top of this page](https://github.com/zeit/hyper/pull/1772#issue-225013615)?

arvindch commented:
Is there any update on this? I would like to be able to disable auto-reload, as well.
